libs/util/git/src/git-client.service.ts

```diff
@@ -297,11 +300,17 @@ export class GitClientService {
 
     // On top of the previous amplication build, apply the patch for the already existing changes that have been in this branch
     if (preCommitDiff.diff) {
+      const userCommitMessages = await this.getUserCommits(
+        tempBranchName,
+        gitCli
+      );
+      await gitCli.checkout(branchName);
```

Comment:
This looks a nice solution but I'd suggest to move the logic within the [const preCommitDiff = await this.calculateDiffAndResetBranch](https://github.com/amplication/amplication/pull/7574/files#diff-7030ed1d7460a9345852066fcdbb545162d9aa94cf278b82f9d0e5eebe1a9600R284) method to avoid unwanted behviours caused by new changes between these two areas of the code.
In addition, `calculateDiffAndResetBranch` already manipulate the local state of the git repository making it a more suitable place for this change.